### PR TITLE
Add Module Docstrings

### PR DESCRIPTION
**Pull request by ToluClassics**

### CODE REVIEW

File: skllm/models/gpt_zero_shot_clf.py [hunk 1]
```diff
@@ -19,6 +19,18 @@
...
 class _BaseZeroShotGPTClassifier(ABC, BaseEstimator, ClassifierMixin, _OAIMixin):
+    """
+    A base class for zero-shot classification using GPT-3.
+
+    Initialization Parameters
+    ----------
+    openai_key : str, optional
+        The OPEN AI key to use. Defaults to None.
+    openai_org : str, optional
+        The OPEN AI organization to use. Defaults to None.
```
Comment (kitsamho) [inline skllm/models/gpt_zero_shot_clf.py:30]: I would change this to `The OPEN AI organization ID`

Comment (ToluClassics) [reply skllm/models/gpt_zero_shot_clf.py:30]: I see, that's still a string right?

**Review by OKUA1 — CHANGES_REQUESTED** [review]
Please see the comments

File: skllm/openai/base_gpt.py [hunk 2]
```diff
@@ -31,10 +44,37 @@ def _get_chat_completion(self, X):
             print(f"Skipping a sample due to the following error: {str(e)}")
             return self.default_output
             
-    def fit(self, X: Any = None, y: Any = None, **kwargs: Any):
+    def fit(self, X: Any = None, y: Any = None, **kwargs: Any) -> "_BaseZeroShotGPTTransformer":
```
Comment (iryna-kondr) [inline skllm/openai/base_gpt.py:47]: Please change the return type

File: skllm/gpt4all_client.py [hunk 3]
```diff
@@ -6,18 +8,34 @@
 _loaded_models = {}
 
 
-def get_chat_completion(messages, model="ggml-gpt4all-j-v1.3-groovy"):
+def get_chat_completion(messages: Dict, model: str="ggml-gpt4all-j-v1.3-groovy") -> Dict:
+    """
+    Get a chat completion from GPT4All which Format list of message dictionaries into a prompt and call model
```
Comment (iryna-kondr) [inline skllm/gpt4all_client.py:13]: `Get a chat completion from GPT4All` would be enough

File: skllm/openai/chatgpt.py [hunk 4]
```diff
@@ -1,19 +1,54 @@
...
+    -------
+    message : dict
+    """
     if role not in ("system", "user", "assistant"):
         raise ValueError("Invalid role")
     return {"role": role, "content": content}
 
 
-def get_chat_completion(messages, key, org, model="gpt-3.5-turbo", max_retries=3):
+def get_chat_completion(messages: str, key: str, org: str, model: str="gpt-3.5-turbo", max_retries: int=3):
```
Comment (iryna-kondr) [inline skllm/openai/chatgpt.py:31]: Wrong type: `messages` is a dict

File: skllm/models/gpt_zero_shot_clf.py [hunk 5]
```diff
@@ -60,11 +73,35 @@ def fit(
         X: Optional[Union[np.ndarray, pd.Series, List[str]]],
         y: Union[np.ndarray, pd.Series, List[str], List[List[str]]],
     ):
+        """
+        Fits the model by storing the training data and extracting the unique targets.
```
Comment (iryna-kondr) [inline skllm/models/gpt_zero_shot_clf.py:77]: The training data is not stored, only the labels are extracted.

**Review by iryna-kondr — CHANGES_REQUESTED** [review]
In addition to the mentioned comments, could you please adjusts most of the docstrings such that they are written in the third person, singular form (e.g. "Fits the model")?

**Review by OKUA1 — APPROVED** [review]

**Pull request merged**